src/test/operators/table_scan_sorted_test.cpp

```diff
@@ -0,0 +1,175 @@
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+#include "operators/limit.hpp"
+#include "operators/table_wrapper.hpp"
+
+namespace opossum {
+
+using Params = std::tuple<EncodingType, bool, std::pair<PredicateCondition, std::vector<AllTypeVariant>>, DataType,
```

Comment:
Your code does not depend on EncodingType, the DataType and whether the segment is a reference segment. I'd leave those dimensions out of this test.

Reply:
Tests are now overhauled and way less complex :+1: